core/services/keeper/orm.go

```diff
@@ -135,9 +143,13 @@ func (korm ORM) SetLastRunHeightForUpkeepOnJob(db *gorm.DB, jobID int32, upkeepI
 		).Error
 }
 
-func (korm ORM) CreateEthTransactionForUpkeep(tx *gorm.DB, upkeep UpkeepRegistration, payload []byte) (bulletprooftxmanager.EthTx, error) {
+func (korm ORM) CreateEthTransactionForUpkeep(ctx context.Context, upkeep UpkeepRegistration, payload []byte) (bulletprooftxmanager.EthTx, error) {
 	from := upkeep.Registry.FromAddress.Address()
 	to := upkeep.Registry.ContractAddress.Address()
 	gasLimit := upkeep.ExecuteGas + korm.config.KeeperRegistryPerformGasOverhead()
-	return korm.txm.CreateEthTransaction(tx, from, to, payload, gasLimit, nil, korm.strategy)
+	return korm.txm.CreateEthTransaction(korm.getDB(ctx), from, to, payload, gasLimit, nil, korm.strategy)
+}
+
+func (korm ORM) getDB(ctx context.Context) *gorm.DB {
+	return postgres.TxFromContext(ctx, korm.DB)
```

Comment:
It retrieves a transaction from the context, or uses the regular DB connection to perform a query